snapcraft/plugins/catkin.py

```diff
@@ -139,6 +139,11 @@ def env(self, root):
             # and use $ROS_HOME to determine where to put them.
             'ROS_HOME=$SNAP_USER_DATA/ros',
 
+            # ROS seems to bust with locale errors if used on anything other
+            # than C or C.UTF-8, so we'll set that here for a consistent
+            # experience across Ubuntu Core and Desktop environments.
```

Comment:
A [boost one](https://svn.boost.org/trac/boost/ticket/4688) actually. They claim it should be fixed in a newer version, but of course, that doesn't help us here.